libglusterfs/src/xlator.c

```diff
@@ -472,6 +472,27 @@ xlator_set_inode_lru_limit(xlator_t *this, void *data)
     return;
 }
 
+void
+xlator_set_inode_table_size(xlator_t *this, void *data)
+{
+    int inode_table_size = 0;
+
+    if (this->itable) {
+        if (!data) {
+            gf_smsg(this->name, GF_LOG_WARNING, 0, LG_MSG_INPUT_DATA_NULL,
+                    NULL);
+            goto out;
+        }
+        inode_table_size = *(int *)data;
+        //inode_table_set_table_size(this->itable, inode_table_size);
+        /* can we replace the table ? */
+        /* alternatively, create a new table and migrate all entries? */
```

Comment:
I'm thinking the *simplest* approach is probably to in xlator_set_inode_table_size just create a new empty table and discard the old one, let it rebuild.  This setting isn't something that should be changed lightly, and I *suspect* the overall impact of having the table rebuild over time is probably lower (and less risky) in the instant than trying to migrate data ... so we're going to be degraded for a bit whilst the table rebuilds, but we don't block (potentially for seconds as a whole) whilst this rebuild happens.

So:

```
if (inode_table_size != this->itable->inode_hashsize) {
    inode_table_t* tmp = inode_table_with_invalidator(this->itable->lru_limit,
        this, this->itable->invalidator_fn, this->itable->dentry_hashsize,
        inode_table_size);
    if (tmp->inode_hashsize != this->itable->inode_hashsize) {
        inode_table_destroy(this->itable);
        this->itable = tmp;
    } else {
        inode_table_destroy(tmp);
    }
}
```

Reply:
I went with this idea.  Don't really see anything more sensible. other than to try and move stuff from the one table to the other.